go/api_default.go

```diff
@@ -67,63 +69,99 @@ func handleNotFoundError(w http.ResponseWriter, detail string) {
 	w.Write(payload)
 }
 
-func callGrafana(url string, apiKey string, verb string, payload io.Reader) (int, []byte, error) {
+func callGrafana(admin bool, url string, auth string, verb string, payload io.Reader) (int, []byte, error) {
 	var statusCode int
 	var body []byte
 	var err error
+
+	if os.Getenv("DEBUG") == "true" {
+		logrus.Infoln("---DEBUG request logs---")
+		logrus.Infoln("URL: " + url)
+		logrus.Infoln("Method: " + verb)
+		if payload != nil {
+			buf := new(bytes.Buffer)
+			buf.ReadFrom(payload)
+			s := buf.String()
+			logrus.Infoln("Payload:" + s)
+		}
+	}
+
 	req, err := http.NewRequest(verb, url, payload)
-	req.Header.Set("Authorization", "Bearer"+" "+apiKey)
+
+	var authKey string
+	if admin {
+		if os.Getenv("DEBUG") == "true" {
+			logrus.Debugln("Using basic auth")
+		}
+		authKey = "Basic"
+	} else {
+		if os.Getenv("DEBUG") == "true" {
+			logrus.Debugln("Using bearer auth")
+		}
+		authKey = "Bearer"
+	}
+
+	req.Header.Set("Authorization", authKey+" "+auth)
 	req.Header.Set("Accept", "application/json")
 	req.Header.Set("Content-Type", "application/json")
 	client := &http.Client{}
 	resp, err := client.Do(req)
+
+	if err != nil {
+		logrus.Infoln("Error calling Grafana:", verb, url, err)
```

Comment:
Does that work? ... i thought your have to 

```Go
logrus.Infof("Error calling Grafana: %s %s %s", verb, url, err)
```

Reply:
actually probably does work ..

Reply:
Yeh both work. Not sure of the preferred syntax. Will check.